app/src/main/java/com/github/factotum_sdp/factotum/ui/maps/MapsFragment.kt

```diff
@@ -50,11 +52,8 @@ class MapsFragment : Fragment() {
         mapFragment.getMapAsync{ googleMap ->
             // clears map from previous markers
             googleMap.clear()
-            //Shows destinations of selected roads
-            for (route in viewModel.routes.value.orEmpty()){
-                route.addDstToMap(googleMap)
-            }
-            googleMap.moveCamera(CameraUpdateFactory.newLatLng(EPFL_LOC))
+            // places markers on the map and centers the camera
+            placeMarkers(viewModel.routes.value!!, googleMap)
```

Comment:
Route values could be null and should be handled instead of throwing an exception. Maybe could just check if null before, if not place the Markers.